### PR TITLE
Parse dropped timeseries from Create Timeseries Response

### PR DESCRIPTION
**Pull request by james-bebbington**
See the test case for an example of a partial success response

### CODE REVIEW

File: metrics_batcher_test.go [hunk 1]
```diff
@@ -109,3 +110,21 @@ func makeTs(i int) *monitoringpb.TimeSeries {
...
+func TestSendReqAndParseDropped(t *testing.T) {
+	persistedCreateTimeSeries := createTimeSeries
+	createTimeSeries = func(ctx context.Context, c *monitoring.MetricClient, ts *monitoringpb.CreateTimeSeriesRequest) error {
+		return errors.New("One or more TimeSeries could not be written: Internal error encountered. Please retry after a few seconds. If internal errors persist, contact support at https://cloud.google.com/support/docs.: timeSeries[0-16,25-44,46-74]; Unknown metric: agent.googleapis.com/system.swap.page_faults: timeSeries[45]")
+	}
+
+	mc, _ := monitoring.NewMetricClient(context.Background())
+	d, err := sendReq(context.Background(), mc, nil)
+	if err == nil {
+		t.Fatalf("Expected err")
```
Comment (nilebox) [inline metrics_batcher_test.go:123]:
```suggestion
		t.Fatal("Expected err")
```

File: metrics_batcher.go [hunk 2]
```diff
@@ -131,16 +133,45 @@ func (mb *metricsBatcher) sendReqToChan() {
...
+	if c == nil {
+		return 0, nil
+	}
+
+	err := createTimeSeries(ctx, c, req)
+	if err == nil {
+		return 0, nil
+	}
+
+	droppedTimeSeriesRangeMatches := timeSeriesErrRegex.FindAllStringSubmatch(err.Error(), -1)
```
Comment (nilebox) [inline metrics_batcher.go:152]: Shall we match for error message text first, e.g. check if it starts with "One or more TimeSeries could not be written"

File: metrics_batcher_test.go [hunk 3]
```diff
@@ -109,3 +110,21 @@ func makeTs(i int) *monitoringpb.TimeSeries {
 		},
 	}
 }
+
+func TestSendReqAndParseDropped(t *testing.T) {
```
Comment (nilebox) [inline metrics_batcher_test.go:114]: Can you add a test where error doesn't include dropped time series? Or is that already covered by existing tests?

**Review by nilebox — APPROVED** [review]

File: metrics_batcher_test.go (hunk 3 above)
Comment (james-bebbington) [reply metrics_batcher_test.go:114]: The existing tests are not great tbh. I've added tests for all cases related to the code I added

**Pull request merged**